### PR TITLE
Issue curl call to route

### LINKED ISSUE #813: curl call to route
## Is your feature request related to a problem? Please describe

Often tools spit out curl calls instead of useable information for us.

## Describe the solution you'd like

create a binary in the cli, that can take a curl call and convert it to a route. It should be interactive.

### PR DESCRIPTION
**Pull request by owais34**
# The Pull Request is ready

- [x] fixes #813 
- [ ] all actions are passing
- [x] only fixes a single issue

## Overview

Added a cli binary tool to convert a generic curl call to a route file.

## Review points

Convert the already working code , to a cleaner design pattern.

## CLI

- [ ] the change works with both supported node versions
- [ ] the default behaviour did not change
- [ ] shared code has been extracted in a different file

### CODE REVIEW

File: cli/bin/generate-benchmark-from-curl.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env node
+import create from '../src/cli/har-route-builder.js';
```
Comment (Idrinth) [inline cli/bin/generate-benchmark-from-curl.js:2]: wrong file

File: cli/package.json [hunk 2]
```diff
@@ -60,6 +60,7 @@
     "make-benchmark-project": "bin/make-benchmark-project.js",
     "generate-benchmark-from-open-api": "bin/generate-benchmark-from-open-api.js",
     "generate-benchmark-from-har": "bin/generate-benchmark-from-har.js",
+    "generate-benchmark-from-curl": "bin/generate-benchmark-from-curl.js",
```
Comment (Idrinth) [inline cli/package.json:63]: please also add a short form below, following the usual rules

File: cli/src/cli/curl-route-builder.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  },
+  '-H': (value: string, routeParams: RouteParams,) => {
+    const headerValues = value.split(':',);
+    // eslint-disable-next-line no-magic-numbers
+    if (headerValues.length !== 2) {
+      throw new Error(language('invalid_request_property', '', 'header',),);
+    }
+    // eslint-disable-next-line no-magic-numbers
+    routeParams.headerSetter=routeParams.headerSetter.concat(
+      // eslint-disable-next-line max-len, no-magic-numbers
```
Comment (Idrinth) [inline cli/src/cli/curl-route-builder.ts:78]: Why not use a constant here?

File: cli/src/cli/curl-route-builder.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    url: '',
+    methodSetter: 'requestInit.method = \'GET\';\n',
+  };
+  // eslint-disable-next-line no-magic-numbers
+  for (let argIndex=FIRST_ARGUMENT; argIndex<args.length-1; argIndex+=2) {
+    argumentHandlers[args[argIndex]](args[argIndex+NEXT], routeParams,);
+  }
+
+  // eslint-disable-next-line no-magic-numbers
+  routeParams.url = args[args.length-1];
```
Comment (Idrinth) [inline cli/src/cli/curl-route-builder.ts:103]: Why not use a constant here?

File: cli/src/cli/curl-route-builder.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+  );
+  stream.write(
+    `import {
+  Task,
+} from '@idrinth/api-bench';
+
+const requestInit : RequestInit = {};
+${ routeParams.methodSetter }
+${ routeParams.requestInitSetter }
+const request: Request = new Request('${ routeParams.url }', requestInit,);
```
Comment (Idrinth) [inline cli/src/cli/curl-route-builder.ts:36]: Request is not imported from anywhere - why not make this a simple object?

**Review by Idrinth — CHANGES_REQUESTED** [review]
Looks pretty good already!

**Comment by owais34** [comment]
@Idrinth , i'll fix the above issues  👍

**Review by Idrinth — DISMISSED** [review]

**Pull request merged**